internal/services/cdn/cdn_endpoint_custom_domain_resource.go

```diff
@@ -159,3 +351,178 @@ func resourceArmCdnEndpointCustomDomainDelete(d *pluginsdk.ResourceData, meta in
+			UpdateRule:        utils.String("NoAction"),
+			DeleteRule:        utils.String("NoAction"),
+		},
+		CertificateSource: cdn.CertificateSourceAzureKeyVault,
+		ProtocolType:      cdn.ProtocolTypeServerNameIndication,
+		MinimumTLSVersion: cdn.MinimumTLSVersionNone,
+	}
+
+	return output, nil
+}
+
+func flattenArmCdnEndpointCustomDomainCdnManagedHttpsSettings(input cdn.ManagedHTTPSParameters) []interface{} {
+	certificateType := ""
+	if params := input.CertificateSourceParameters; params != nil {
+		certificateType = string(params.CertificateType)
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"certificate_type": certificateType,
+			"protocol_type":    string(input.ProtocolType),
+		},
+	}
+}
+
+func flattenArmCdnEndpointCustomDomainUserManagedHttpsSettings(input cdn.UserManagedHTTPSParameters) []interface{} {
+	var (
+		subscriptionId    string
+		resourceGroupName string
+		vaultName         string
+		secretName        string
+		secretVersion     string
+	)
+	if params := input.CertificateSourceParameters; params != nil {
+		if params.SubscriptionID != nil {
+			subscriptionId = *params.SubscriptionID
+		}
+		if params.ResourceGroupName != nil {
+			resourceGroupName = *params.ResourceGroupName
+		}
+		if params.VaultName != nil {
+			vaultName = *params.VaultName
+		}
+		if params.SecretName != nil {
+			secretName = *params.SecretName
+		}
+		if params.SecretVersion != nil {
+			secretVersion = *params.SecretVersion
+		}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"key_vault_id":   keyvaultParse.NewVaultID(subscriptionId, resourceGroupName, vaultName).ID(),
+			"secret_name":    secretName,
+			"secret_version": secretVersion,
+		},
+	}
+}
+
+func enableArmCdnEndpointCustomDomainHttps(ctx context.Context, client *cdn.CustomDomainsClient, id parse.CustomDomainId, params cdn.BasicCustomDomainHTTPSParameters) error {
+	_, err := client.EnableCustomHTTPS(ctx, id.ResourceGroup, id.ProfileName, id.EndpointName, id.Name, &params)
+	if err != nil {
+		return fmt.Errorf("sending enable request: %+v", err)
+	}
+
+	log.Printf("[DEBUG] Waiting for HTTPS to enable on %q", id)
+	deadline, _ := ctx.Deadline()
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{string(cdn.CustomHTTPSProvisioningStateEnabling)},
+		Target:     []string{string(cdn.CustomHTTPSProvisioningStateEnabled), string(cdn.CustomHTTPSProvisioningStateFailed), string(cdn.CustomHTTPSProvisioningStateDisabled)},
+		Refresh:    cdnEndpointCustomDomainHttpsRefreshFunc(ctx, client, id),
+		MinTimeout: 10 * time.Second,
+		Timeout:    time.Until(deadline),
+	}
+
+	state, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("waiting for HTTPS provision state: %+v", err)
+	}
+	if state == cdn.CustomHTTPSProvisioningStateFailed {
+		return errors.New("HTTPS provision state is Failed")
+	}
+	if state == cdn.CustomHTTPSProvisioningStateDisabled {
+		return errors.New("HTTPS provision state is back to Disabled")
```

Comment:
(as below) two things, 1:

```suggestion
		return fmt.Errorf("HTTPS provision state is back to Disabled")
```

2: isn't this an invalid state, so shouldn't this be removed?